app/src/main/java/com/neiapp/spocan/ui/activity/InitiativeActivity.java

```diff
@@ -23,28 +38,91 @@ protected void onCreate(Bundle savedInstanceState) {
         mAddPhotoBtn = findViewById(R.id.addPhotoBtn);
         mCancelPublishBtn = findViewById(R.id.cancel_button);
         mPublishInitiativeBtn = findViewById(R.id.publish_button);
-
+        textDescription = findViewById(R.id.editTextDescription);
+        imageView = findViewById((R.id.imageView));
 
         mAddPhotoBtn.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Log.d( "addPhoto", "Aca se deberia pasar a la pantalla que agrega la foto");
+                if (ContextCompat.checkSelfPermission(getApplicationContext(), Manifest.permission.CAMERA) == PackageManager.PERMISSION_DENIED) {
+                    final String[] permission = {Manifest.permission.CAMERA};
+                    requestPermissions(permission, 1001);
+                } else {
+                    Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+                    startActivityForResult(intent, 1001);
+                    setResult(RESULT_OK);
+                }
             }
         });
 
         mCancelPublishBtn.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Log.d( "cancel", "Aca se deberia volver al home o desde donde se hubiera llamado a agregar iniciativa");
+                Intent intent = new Intent(InitiativeActivity.this, MainActivity.class);
+                startActivity(intent);
             }
         });
 
         mPublishInitiativeBtn.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                Log.d( "publish", "Aca se deberia mostrar un toast con _operacion completada_ y redirigirse al home/muro");
+                if (validateField()) {
+                    Initiative intitiative = new Initiative(textDescription.getText().toString(), bitmap, true);
+                    Backend backend = Backend.getInstance();
+                    backend.createInitiative(intitiative, new CallbackVoid() {
+                        @Override
+                        public void onSuccess() {
+                            Intent intent = new Intent(InitiativeActivity.this, MainActivity.class);
```

Comment:
Acá en el InitiativeActivity.this no esta mal, pero es mas seguro (en general) decirle a Android que entienda donde está el proceso actual y a partir de ahi te redirija hacia donde le decis. Con lo que, se podría modificar a "getApplicationContext()"

Reply:
Recordaba lo del "getApplicationContext() pero Nico me habia dicho que InitiativeActivity.this tmb funcionaba y lo cambié por esta otra. Pero ya lo cambio :)